lib/config/yml_generator_config.dart

```diff
@@ -89,62 +89,131 @@ class YmlGeneratorConfig {
         ));
         return;
       }
-      if (properties == null) {
+      if (properties == null && type != 'enum') {
         throw Exception('Properties can not be null. model: $key');
       }
-      if (properties is! YamlMap) {
+      if (properties is! YamlMap?) {
         throw Exception(
             'Properties should be a map, right now you are using a ${properties.runtimeType}. model: $key');
       }
       if (type == 'enum') {
-        final uppercaseEnums =
-            (value['uppercase_enums'] ?? pubspecConfig.uppercaseEnums) == true;
-        final itemType = value['item_type'] == null
-            ? const StringType()
-            : _parseSimpleType(value['item_type']);
+        final deprecatedItemType = value['item_type'];
+        if (deprecatedItemType != null) {
+          throw Exception(
+              'item_type is removed, follow the migration to version 7.0.0');
+        }
 
-        if (itemType is! StringType &&
-            itemType is! IntegerType &&
-            itemType is! DoubleType) {
+        final deprecatedGenerateExtensions = value['generate_extensions'];
+        if (deprecatedGenerateExtensions != null) {
           throw Exception(
-              'item_type should be a string or integer. model: $key');
+              'generate_extensions is removed, follow the migration to version 7.0.0');
         }
 
+        final deprecatedGenerateMap = value['generate_map'];
```

Comment:
Why is this deprecated? Because the map is still needed. but it should use the values fields from the enum values instead. Otherwise we will have to write our own mapping each time we want to map something. Keeping this logic will make it the same for all projects